x-pack/plugins/cases/public/components/cases_context/state/use_is_add_to_case_open.test.tsx

```diff
@@ -0,0 +1,105 @@
+import { allCasesPermissions } from '../../../common/mock';
+import { ExternalReferenceAttachmentTypeRegistry } from '../../../client/attachment_framework/external_reference_registry';
+import { PersistableStateAttachmentTypeRegistry } from '../../../client/attachment_framework/persistable_state_registry';
+import { useIsAddToCaseOpen } from './use_is_add_to_case_open';
+import { useApplication } from '../../../common/lib/kibana/use_application';
+import { of } from 'rxjs';
+import { useCasesToast } from '../../../common/use_cases_toast';
+import type { PublicAppInfo } from '@kbn/core/public';
+
+jest.mock('../../../common/use_cases_toast');
+jest.mock('../../../common/lib/kibana');
+jest.mock('../../../common/lib/kibana/use_application');
+jest.mock('../../../common/use_cases_toast');
+jest.mock('../../all_cases/selector_modal/all_cases_selector_modal', () => {
+  return {
+    AllCasesSelectorModal: jest.fn(),
+  };
+});
+
+const externalReferenceAttachmentTypeRegistry = new ExternalReferenceAttachmentTypeRegistry();
+const persistableStateAttachmentTypeRegistry = new PersistableStateAttachmentTypeRegistry();
+const useKibanaMock = useKibana as jest.Mocked<typeof useKibana>;
+const useApplicationMock = useApplication as jest.Mock;
+const useCasesToastMock = useCasesToast as jest.Mock;
+const mockedToastInfo = jest.fn();
+
+describe('use is add to existing case modal open hook', () => {
+  let wrapper: FC<PropsWithChildren<unknown>>;
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    useKibanaMock().services.application = {
+      ...useKibanaMock().services.application,
+      currentAppId$: of('securitySolutionUI'),
+      applications$: of(
+        new Map([
+          ['securitySolutionUI', { category: { label: 'Test' } } as unknown as PublicAppInfo],
+        ])
+      ),
+    };
+    useApplicationMock.mockReturnValue({ appId: 'testAppId' });
+    useCasesToastMock.mockReturnValue({
+      showInfoToast: mockedToastInfo,
+    });
+
+    wrapper = ({ children }) => {
+      return (
+        <CasesProvider
+          value={{
+            externalReferenceAttachmentTypeRegistry,
+            persistableStateAttachmentTypeRegistry,
+            owner: ['cases'],
+            permissions: allCasesPermissions(),
+            basePath: '/jest',
+            features: { alerts: { sync: true, enabled: true, isExperimental: false }, metrics: [] },
+            releasePhase: 'ga',
+            getFilesClient: jest.fn(),
+          }}
+        >
+          {children}
+        </CasesProvider>
+      );
+    };
+  });
+
+  it('should throw if called outside of a cases context', () => {
+    const { result } = renderHook(useIsAddToCaseOpen);
+    expect(result.error?.message).toContain(
+      'useCasesStateContext must be used within a CasesProvider and have a defined value'
+    );
+  });
+
+  it('should return open modal status', async () => {
```

Comment:
nit: Should we also add another test for the flyout?

Reply:
Yup, added test for the flyout open and also for everything closed